server/src/main/java/io/druid/client/SegmentLoadInfo.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.client;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.api.client.repackaged.com.google.common.base.Preconditions;
+import io.druid.server.coordination.DruidServerMetadata;
+import io.druid.timeline.DataSegment;
+
+import java.util.List;
+import java.util.Set;
+
+public class SegmentLoadInfo
+{
+  private final DataSegment segment;
+  private final Set<DruidServerMetadata> servers;
+
+  @JsonCreator
+  public SegmentLoadInfo(
```

Comment:
this is serialized when the realtime task polls about the status of segments loaded via http end point. java doesn't have any ConcurrentSet api which can be used to enforce it. 
Right now only coordinatorServerView creates that object. 
Do you have any suggestions on how we can enforce it ?